### PR TITLE
Disable section pages

### PR DESCRIPTION
**Pull request by AaronCrawfis**
This PR disables section pages.

But wait, why would we do that!?!?

Well, for one the section pages don't offer a ton of value. They have an auto-generated summary of the child pages, but they don't have much content themselves. This causes people to land on one and wonder "ok now what, which link do I choose?"

In this new format, users can click on sections and they will expand to show what's in the section. Then they can choose to look at an overview page detailing the resource, or they can choose one of the sub-pages.

### CODE REVIEW

File: docs/content/author-apps/dev-environment/initialize-environment.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Radius development environments enable you to quickly get started with Radius an
 
 ## How-to: Initialize a new dev environment
 
-1. Initialize a new [Radius environment]{{(< ref "operations/environments">)}} with `rad init --dev` command:
+1. Initialize a new [Radius environment]{{(< ref "operations/environments/overview">)}} with `rad init --dev` command:
```
Comment (willtsai) [inline docs/content/author-apps/dev-environment/initialize-environment.md]:
```suggestion
1. Initialize a new [Radius environment]({{< ref "operations/environments/overview">}}) with `rad init --dev` command:
```

File: docs/content/author-apps/networking/overview/index.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+type: docs
+title: "Overview: Application networking"
+linkTitle: "Network communication"
```
Comment (willtsai) [inline docs/content/author-apps/networking/overview/index.md]: Just a preference, not a hard suggestion, but I think I'd prefer "Application networking" as the side bar title to better match the page title:

```suggestion
linkTitle: "Application networking"
```

File: docs/content/operations/environments/workspaces/_index.md
Comment (willtsai) [inline docs/content/operations/environments/workspaces/_index.md:1]: should this be renamed to `index.md` ?

File: docs/content/operations/providers/_index.md
Comment (willtsai) [inline docs/content/operations/providers/_index.md:1]: should this be renamed to `index.md` ?

File: docs/content/reference/resource-schema/core-schema/volumes/volume-azure-keyvault/_index.md
Comment (willtsai) [inline docs/content/reference/resource-schema/core-schema/volumes/volume-azure-keyvault/_index.md:1]: should this be renamed to `index.md` ?

**Review by willtsai — CHANGES_REQUESTED** [review]

File: docs/content/author-apps/networking/overview/index.md (hunk 2 above)
Comment (AaronCrawfis) [reply docs/content/author-apps/networking/overview/index.md]: This is perfect, I was struggling on the term yesterday 😅

**Review by willtsai — APPROVED** [review]

**Pull request merged**